### PR TITLE
added check for username and email

### PR DESCRIPTION
**Pull request by niksm7**
## Related Issue or bug
The register form expects username and email which should be unique according to Django authentication but when some other user tries to register with already existing username or email they only come to know about the problem after submitting the form which is not good user experience.

Fixes: #226 

#### Describe the changes you've made

I have added Ajax functionality to dynamically check the uniqueness of the username and email while registering. The Ajax call checks if the user with entered name or email exists and if it is so then the user is alerted with a message. Even after showing the error if the user tries to register then they are shown a window alert  stating the problem. Also if the user doesn't match the password and confirm password field then they are alerted about the same. This functionality of uniqueness is also maintained with the google login as well.

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I have tested it against various scenarios and it works perfectly fine also shown in the video below

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

## Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

## Screenshots

https://user-images.githubusercontent.com/55614782/114166703-b8629f00-994b-11eb-8bd1-5e61b3c11543.mov

### CODE REVIEW

**Comment by swapnilsparsh** [comment]
@niksm7 Great work !!
Just add a feature that if username/email is available then it should say username/email is available with a green tick  and if it's not available then it should say not available with a cross symbol.

**Review by swapnilsparsh — CHANGES_REQUESTED** [review]
Just add a feature that if username/email is available then it should say username/email is available with a green tick and if it's not available then it should say not available with a cross symbol.

**Comment by niksm7** [comment]
Have added the requested feature @swapnilsparsh 
Screenshot 2021-04-11 at 3 04 42 PM

File: main_app/templates/main_app/register.html [hunk 1]
```diff
@@ -213,18 +239,28 @@ <h3 class="register-sign-in-text">Already on Rescue? <a href="/login">Sign In</a
...
-                            if($("#username_alert").length == 0){
-                                var newNode = document.createElement('small');
-                                newNode.id = "username_alert"
-                                newNode.style = "color: red;font-size: 17px;"
-                                newNode.innerHTML = "username already exists <br>"
-                                var referenceNode = document.querySelector("#id_username");
-                                referenceNode.after(newNode);
-                            }
+                                x.style = "color: red;font-size: 17px;"
+                                x.innerHTML = "<br>username not available <br>"
```
Comment (swapnilsparsh) [inline main_app/templates/main_app/register.html:248]: Make the first letter capital

Comment (swapnilsparsh) [inline main_app/templates/main_app/register.html:248]: Make the first letter capital

File: main_app/templates/main_app/register.html [hunk 2]
```diff
@@ -213,18 +239,28 @@ <h3 class="register-sign-in-text">Already on Rescue? <a href="/login">Sign In</a
...
+                                y.innerHTML = "x"
+                                y.style = "color: red;margin-left: -23px;"
                         }
                         else{
-                            $("#username_alert").remove()
+                            // If the input field is not empty and the username is available
+                            if($('#id_username').val() != ""){
+                                x.style = "color: green;font-size: 17px;"
+                                x.innerHTML = "<br>username is available <br>"
```
Comment (swapnilsparsh) [inline main_app/templates/main_app/register.html:256]: Make the first letter capital

**Review by swapnilsparsh — CHANGES_REQUESTED** [review]
Resolve the conflicts

**Comment by niksm7** [comment]
Changes have been done and conflicts resolved @swapnilsparsh

**Comment by sakshikhachane** [comment]
Hey @niksm7,
Still, there are conflicts. Please resolve them.

**Comment by niksm7** [comment]
Done @sakshikhachane

**Pull request merged**